src/java_binding/src/lib.rs

```diff
@@ -196,7 +196,7 @@ where
     F: FnOnce() -> Result<Ret>,
     Ret: Default,
 {
-    match catch_unwind(std::panic::AssertUnwindSafe(inner)) {
+    match rw_catch_unwind(std::panic::AssertUnwindSafe(inner)) {
```

Comment:
This code here is not within the context of a tokio runtime. I am not sure whether it will fail when accessing the tokio task local variable.

Reply:
`tokio::task_local` has nothing specific to `tokio` actually, 😂 which is verified in the unit test of `test_sync_rw`. I'm wishing there could be a standalone crate to provide the scope-local functionality. 🤪